src/MSBuildBinLogQuery/Graph/IQueryableGraphNode.cs

```diff
@@ -1,6 +1,13 @@
-﻿namespace Microsoft.Build.Logging.Query.Graph
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Build.Logging.Query.Graph
 {
     public interface IQueryableGraphNode
     {
     }
+
+    public interface IQueryableGraphNode<T> : IQueryableGraphNode where T : IQueryableGraphNode
+    {
+        ISet<T> AdjacentNodes { get; }
```

Comment:
No concrete suggestions here, but I'd like to make sure you look at some prior art in API design:

* [`System.Xml.Linq.XElement`](https://docs.microsoft.com/en-us/dotnet/api/system.xml.linq.xelement?view=netframework-4.8) has `Parent`/`PreviousNode`/`NextNode`/`FirstNode` where `FirstNode` is `FirstChildNode`.
* [`System.Xml.XmlNode`](https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmlnode?view=netframework-4.8) has `ParentNode`/`PreviousSibling`/`NextSibling`/`ChildNodes` (I like these names better).

Also relevant: [`IXpathNavigable`](https://docs.microsoft.com/en-us/dotnet/api/system.xml.xpath.ixpathnavigable?view=netcore-3.0).